app/services/attribute_uri_converter.rb

```diff
@@ -0,0 +1,47 @@
+class AttributeURIConverter
+  def initialize(attributes)
+    @attributes = attributes.dup
+  end
+
+  def convert_attributes
+    @attributes.each do |property, values|
+      convert_attribute(property, values)
+    end
+
+    @attributes
+  end
+
+  private
+
+  def convert_attribute(property, values)
+    case values
```

Comment:
I was more concerned about a hash, since nested attributes could be a thing in pretty much any Rails form.  I could force it to an array, but we don't want hashes to get hit with this logic.

We can definitely merge string and array cases, but I don't know offhand how to reject the hash possibility.